docs/decisions/20240604-hdkey-domain-separation

```diff
@@ -0,0 +1,62 @@
+# HDKey determinstic key and domain derivation when using different algorithms
+
+- Status: in-progress 
+- Deciders: Javi, Yurii, Jesus, Matt, Ezequiel
+- Date: 2024-06-04
+- Tags: Domain separation, Deterministic wallets
+
+
+## Context and Problem Statement
+
+The main question to answer is: How can we add ed25519 HDKey deterministic wallet implementation on top of existing secp256k1 HDKey implementation and make it imposssible for the users to generate the same keys using the same seed?
+
+For broader context, we are now introducing ed25519 HDKey deterministic wallet with Cardanos bip32 implementation for ed25519 and this will work together with our existing standard bip32 implementation which is used by secp256k1.
+
+This deterministic keys are going to be used by the issuer to issue credentials with both algorithms and it is important to mention, right now the Bip32 is deriving using the following paths.
+
+Currently in Bip32 (secp256k1), the path looks like this m/0'/0'/0' and we are just incrementing the accountIndex whenever it is needed.
+There is a risk now where user's could potentially be creating the same keys using different key curves and find a colision. 
```

Comment:
I am not entirely following the problem definition.
Secp256k1 keys represent elements on the secp256k1 curve, while ed25519 keys are (more or less) elements on the Curve25519 curve. Unless I am misinterpreting something, I am not following how a collision could occur between different curves (unless they actually share points, but I don't think so in this case)

I do see a potential issue for collisions within the same curve if we use the same seed to derive keys for diferent purposes. In the original derivation scheme for secp keys, we divided the domain so that keys used for signing could not match keys used for encryption. 
For Edwards's keys, there is likely similar security considerations, and a potential sharing of Cardano wallet's seed and our seed could lead to such scenarios. We can use a deviation from bip32 to separate the space with a level of the derivation tree. However, this does not necesarilly need to change secp current derivation paths (which if I recall correctly was already not in conflict with bip32 wallets for cryptocurrencies, but I can be wrong on this)

Reply:
@jesusdiazvico Can you maybe clarify

Reply:
@EzequielPostan @jesusdiazvico is this https://staging-docs.atalaprism.io/adrs/adr/20230516-hierarchical-deterministic-key-generation-algorithm/ 

I see this ADR that u guys worked on is accepted why are we not doing this already :) I'm little lost. then let's just do it. I can close the current ADR and just implement what we already agreed!